sky/clouds/service_catalog/common.py

```diff
@@ -46,15 +48,90 @@ def get_catalog_path(filename: str) -> str:
     return os.path.join(_CATALOG_DIR, filename)
 
 
-def read_catalog(filename: str) -> pd.DataFrame:
-    """Reads the catalog from a local CSV file.
+def _get_filtered_cache_path(filename: str, area_filter: List[str]) -> str:
+    """Returns the cache path for the filtered catalog.
 
-    If the file does not exist, download the up-to-date catalog that matches
-    the schema version.
+    Args:
+        filename: The filename of the catalog.
+        area_filter: The area filter.
+
+    Returns:
+        The cache path.
+    """
+    assert filename.endswith('.csv'), filename
+    new_name = filename.split('.csv')[0] + '.filtered.'
+    cached_catalog_path = get_catalog_path(new_name)
+    cached_catalog_path += f'{"_".join(area_filter)}.csv'
+    return cached_catalog_path
+
+
+def _read_catalog_config() -> Optional[Dict[str, Dict[str, str]]]:
```

Comment:
Nice, eventually we should converge on something like a `SkyConfig` class which exposes methods to read the `config` file.